components/SpotifyInterface/SpotifyEventHandler.c

```diff
@@ -0,0 +1,146 @@
+            if (xQueueReceive((*EventData_t->HttpsBufQueue), TempBuffer, portMAX_DELAY) == pdTRUE)
+            {
+                ESP_LOGI(TAG, "Receive data in Event handler by queue ");
+            }
+            EventData_t->EventHandlerCallBackFunction(TempBuffer);
+            break;
+        }
+        case SpotifyEventSendRequestForPause_:
+        {
+            Spotify_SendRequestForPause(EventData_t->token);
+            if (xQueueReceive((*EventData_t->HttpsBufQueue), TempBuffer, portMAX_DELAY) == pdTRUE)
+            {
+                ESP_LOGI(TAG, "Receive data in Event handler by queue ");
+            }
+            EventData_t->EventHandlerCallBackFunction(TempBuffer);
+            break;
+        }
+        case SpotifyEventGetUserStatus_:
+        {
+            Spotify_GetUserStatus(EventData_t->token);
+            if (xQueueReceive((*EventData_t->HttpsBufQueue), TempBuffer, portMAX_DELAY) == pdTRUE)
+            {
+                ESP_LOGI(TAG, "Receive data in Event handler by queue ");
+            }
+            EventData_t->EventHandlerCallBackFunction(TempBuffer);
+            break;
+        }
+        case SpotifyEventGetUserTopItems_:
+        {
+            Spotify_GetUserTopItems(EventData_t->token);
+            if (xQueueReceive((*EventData_t->HttpsBufQueue), TempBuffer, portMAX_DELAY) == pdTRUE)
+            {
+                ESP_LOGI(TAG, "Receive data in Event handler by queue ");
+            }
+            EventData_t->EventHandlerCallBackFunction(TempBuffer);
+            break;
+        }
+        case SpotifyEventGetUserProfile_:
+        {
+            if ((EventData_t->UserInfo.UserID[0]) == 0)
+            {
+                ESP_LOGE(TAG, "you should run 'GetUserInfo' before this !");
+                free(TempBuffer);
+                return;
+            }
+            Spotify_GetUserProfile(EventData_t->UserInfo.UserID, EventData_t->token);
+            if (xQueueReceive((*EventData_t->HttpsBufQueue), TempBuffer, portMAX_DELAY) == pdTRUE)
+            {
+                ESP_LOGI(TAG, "Receive data in Event handler by queue ");
+            }
+            EventData_t->EventHandlerCallBackFunction(TempBuffer);
+            break;
+        }
+        case SpotifyEventGetCurrentPlaying_:
+        {
+            Spotify_GetCurrentPlaying(EventData_t->token);
+            if (xQueueReceive((*EventData_t->HttpsBufQueue), TempBuffer, portMAX_DELAY) == pdTRUE)
+            {
+                ESP_LOGI(TAG, "Receive data in Event handler by queue ");
+            }
+            EventData_t->EventHandlerCallBackFunction(TempBuffer);
+            break;
+        }
+        }
+    }
+    free(TempBuffer);
+}
+
+/**
+ * @brief This function register event handler for spotify , this event is task and need
+ *        long stack ,because spotify functions need long stack
+ */
+void Spotify_RegisterEventHandler(void)
+{
+    esp_event_loop_args_t Spotify_EventLoopArgs = {
+        .queue_size = 5,
```

Comment:
why 5? use a Macro instead of the number

Reply:
I change it to "#define EVENT_LOOP_QUEUE 5"